### ISSUE TITLE
Problem running QZXingLive Example

### ISSUE THREAD

**Issue by kpkale** [ftylitak/qzxing#48]
Hello
I am just unable to run the QZXingLive Example.
I have been following up issues related to this example in this repository.
I am using Qt5.7 on Ubuntu with Latest Android Build Sdk : android-26.
Tried using gradle.. i get this error.

> Warning: QML import could not be resolved in any of the import paths: QZXing
> Warning: QML import could not be resolved in any of the import paths: QtQuick.Controls.Material.impl
> Warning: QML import could not be resolved in any of the import paths: QtQuick.Controls.impl
> Warning: QML import could not be resolved in any of the import paths: QtQuick.Controls.Universal.impl
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
> Build file '/home/kaushik/QtProjects/QZingLive-Android-Release/android-build/build.gradle' line: 17
> 
> * What went wrong:
> A problem occurred evaluating root project 'android-build'.
> > java.lang.UnsupportedClassVersionError: com/android/build/gradle/AppPlugin : Unsupported major.minor version 52.0
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
> 
> BUILD FAILED 

Tried to build using ant but get this error.
BUILD FAILED
/home/kaushik/Android/android-sdk-linux/tools/ant/build.xml:649: The following error occurred while executing this line:
/home/kaushik/Android/android-sdk-linux/tools/ant/build.xml:694: null returned: 139

I just need to get this example working so that I can use this in my application.

Thanks for your time.

**Comment by MohammadKanan** [comment]
Hi,
A suggestion: since you clearly got the error pointing to line 17 of your  "build.gradle" , it might help if you show what you have in that section.
or you might want to compare your build.gradle with one that do not generate errors, like blow:
```
buildscript {
    repositories {
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:2.2.2'
    }
}

allprojects {
    repositories {
        jcenter()
    }
}

apply plugin: 'com.android.application'

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    compile 'com.android.support:appcompat-v7:25.0.1'
    compile 'com.android.support:design:25.0.1'
}

android {
    /*******************************************************
     * The following variables:
     * - androidBuildToolsVersion,
     * - androidCompileSdkVersion
     * - qt5AndroidDir - holds the path to qt android files
     *                   needed to build any Qt application
     *                   on Android.
     *
     * are defined in gradle.properties file. This file is
     * updated by QtCreator and androiddeployqt tools.
     * Changing them manually might break the compilation!
     *******************************************************/

    compileSdkVersion androidCompileSdkVersion.toInteger()

    buildToolsVersion androidBuildToolsVersion

    sourceSets {
        main {
            manifest.srcFile 'AndroidManifest.xml'
            java.srcDirs = [qt5AndroidDir + '/src', 'src', 'java']
            aidl.srcDirs = [qt5AndroidDir + '/src', 'src', 'aidl']
            res.srcDirs = [qt5AndroidDir + '/res', 'res']
            resources.srcDirs = ['src']
            renderscript.srcDirs = ['src']
            assets.srcDirs = ['assets']
            jniLibs.srcDirs = ['libs']
       }
    }

    lintOptions {
        abortOnError false
    }
}
```

**Comment by kpkale** [comment]
Thanks for responding.

I checked the file that you sent and on my PC. They are exactly the same. 

I even tried compiling on the latest Qt5.9.1 but the same errors.

I think I am not including something missing some step.. I am just opening the project and compiling and expect the project to compile successfully. 

What could be wrong?

**Comment by MohammadKanan** [comment]
@kpkale ,
is line 17 in your gradle.build same? 
which kit you are using ?
coud you try below kit:
`Android for armeabi-v7a (GCC 4.9, Qt 5.9.0 for Android armv7)`

**Comment by kpkale** [comment]
@ftylitak From the posts that are seen in the issues, I can clearly see that there are many who have been able to compile the solution. I just dont get what wrong is happening. Some more information so that you could help.
I am using the following:
Qt 5.9.1 (GCC 5.3.1 20160406 (Red Hat 5.3.1-6), 64 bit), QtCreator 4.3.1
Android for armeabi-v7a (GCC 4.9, Qt 5.9.1 for Android armv7)
QMake Location: /home/kaushik/Qt5.9.1/5.9.1/android_armv7/bin/qmake
android-ndk-r10e
java-7-openjdk-amd64
Android Build SDK = android-26
The Operating System is Ubuntu 64bit.

I am regularly developing my applications on this same setup... I just upgraded the Qt to 5.9.1 thinking that might help and for android-26 to run the permissions patch.

**Comment by kpkale** [comment]
@MohammadKanan I was just writing the post with this information.

**Comment by kpkale** [comment]
@MohammadKanan I assume you are able to compile the example for android.
What is the environment in which you are able to compile it?
Is there something else that needs to be done like installing some other library?

**Comment by MohammadKanan** [comment]
@kpkale ,
I am compiling under windows 10, using MinGW 5.3. I recall I fixed few things when compiled first time, I am sorry I cant be sure what it was, but most likely in build.gradle. one more thing .. I was not able to compile with long path names .. So I configured build to be under c:\tmp.  ... but I assume does not apply to your case as you are using Linux.

**Comment by MohammadKanan** [comment]
@kpkale ,
more changes I made in main.qml:
```
import QtQuick 2.5
import QtQuick.Window 2.0
import QtQuick.Controls 2.0
import QtQuick.Layouts 1.1
import QtMultimedia 5.5
```

**Comment by kpkale** [comment]
@MohammadKanan The imports are already present in mail.qml exactly same

**Comment by MohammadKanan** [comment]
@kpkale 
You might want to check this case : [Getting error when trying to run new project in Android Studio](https://stackoverflow.com/questions/37312404/getting-error-when-trying-to-run-new-project-in-android-studio-2-2-1)
You might also want to check this : [Plugin with id 'com.android.application' not found](https://stackoverflow.com/questions/24795079/error1-0-plugin-with-id-com-android-application-not-found)

**Comment by kpkale** [comment]
@MohammadKanan 
I have upgraded the java to version 8 instead of 7 and my error has changed.
The new error is:
```
File /home/kpkale/.android/repositories.cfg could not be loaded.

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'android-build'.
> Failed to find Platform SDK with path: platforms;android-1

* Try:

BUILD FAILED
```
I am not sure if the first line is a part of the problem but it was there just before the FAILURE: so quoting that as well.

**Comment by kpkale** [comment]
@MohammadKanan 
I might further try to create a windows vm and setup the environment and try building it from windows.

**Comment by MohammadKanan** [comment]
@kpkale 
After upgraded Java, you should flush old user settings .. may be you need to clean up the `/home/kpkale/.android`

**Comment by ftylitak** [comment]
@kpkale

have you managed in the past to successfully compile any Qt project targeting Android? There are many example projects in the Welcome page of Qt Creator to try out. If you are having problems on this may be you could re-setup your complete Qt/Android installation following the documentation : http://doc.qt.io/qt-5/androidgs.html

**Comment by kpkale** [comment]
@ftylitak 
You made my day.
I noticed that though the environment was not hampering my process of building and working with existing apps, it was giving errors with new apps or example apps deployment to a device.
So I set up another environment exactly as per the guide, and lol.. the example got deployed to a device. Now I can look at using the functionality in my application.
Thanks.
@MohammadKanan  Thanks for your help.
